### ISSUE TITLE
Qwen3-14B target training performs much worse than Qwen3-4B

### ISSUE THREAD

**Issue by uygnef** [torchspec-project/TorchSpec#190]
We found that DSpark training works normally with a Qwen3-4B target, but performs
poorly after switching to Qwen3-14B.

<img width="2070" height="812" alt="Image" src="https://github.com/user-attachments/assets/a9e83b91-2e20-4901-b7a0-a3f91d8fd756" />

dspark_softmax_qwen3_14b_control: qwen3 14b target
official_softmax_full_24700: qwen3 4b target

The data and target cache were regenerated using Qwen3-14B, so no Qwen3-4B
responses or hidden states were reused.

Main configuration:

```yaml
target_model: Qwen/Qwen3-14B
target_hidden_layers: [1, 10, 19, 28, 37]
num_draft_layers: 5
block_size: 7
attention_backend: softmax
progressive_target_kv: false

micro_batch_size: 1
global_batch_size: 512
learning_rate: 6e-4
warmup_ratio: 0.04
num_train_epochs: 10
precision: bf16
```

This run is the softmax-only DSpark control: it uses full Target-hidden
conditioning and does not enable KDA, Target-KV, strict-prefix visibility, or
progressive hidden visibility.

Launch command:

```bash
#!/usr/bin/env bash
set -euo pipefail

# A 6-node Ray cluster is already running:
# 2 nodes / 16 GPUs for online Target inference
# 4 nodes / 32 GPUs for Draft training

export RAY_ADDRESS="${RAY_ADDRESS:?}"
export TRAIN_DATA="${TRAIN_DATA:?}"
export EVAL_DATA="${EVAL_DATA:?}"

python -m torchspec.train_entry \
  --config configs/sglang_qwen3_14b_dspark_softmax_control.yaml \
  model.target_model_path=Qwen/Qwen3-14B \
  model.draft_model_config=torchspec/config/dspark_softmax_control_qwen3_14b.json \
  dataset.train_data_path="${TRAIN_DATA}" \
  dataset.eval_data_path="${EVAL_DATA}" \
  training.training_num_nodes=4 \
  training.training_num_gpus_per_node=8 \
  training.micro_batch_size=1 \
  training.draft_accumulation_steps=16 \
  training.num_epochs=10 \
  training.learning_rate=6e-4 \
  training.dspark_progressive_target_kv=false \
  inference.inference_num_gpus=16 \
  inference.inference_num_gpus_per_node=8 \
  inference.inference_batch_size=8 \
  inference.inference_fetch_batch=128
```

The corresponding Qwen3-4B configuration trains normally.

**Comment by yubofredwang** [comment]
Was the 14b training data regenerated using qwen3_14b?

**Comment by yubofredwang** [comment]
# Qwen3-4B
lm_head_key: model.embed_tokens.weight

# Qwen3-14B
lm_head_key: lm_head.weight

Qwen3-14B is a special case that does not do tie_word_embeddings
